src/Stratis.Bitcoin/Interfaces/IInitialBlockDownloadState.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Stratis.Bitcoin.Interfaces
+{
+    /// <summary>
+    /// Provides IBD (Initial Block Download) state.
+    /// </summary>
+    public interface IInitialBlockDownloadState
+    {
+        /// <summary>
+        /// Checks whether the node is currently in the process of initial block download.
+        /// </summary>
+        /// <returns><c>true</c> if the node is currently doing IBD, <c>false</c> otherwise.</returns>
+        bool IsInitialBlockDownload();
+
+        /// <summary>
+        /// Sets last IBD status update time and result.
```

Comment:
If its used in tests why does this method need to live on the interface?
Tests can easily override the concrete implementation.

Reply:
It's for the integration tests and inherited from the old implementation- also don't like it being here. 
I've tried to easily remove that but it looks like we need to build a node in integration tests environment with a mock so I decided to do that in a separate PR - too much to change to remove that.